source/material_model/viscoelastic.cc

```diff
@@ -0,0 +1,700 @@
+                             "for a total of N+1 values, where N is the number of compositional fields."
+                             "If only one value is given, then all use the same value. Units: $J /kg /K$");
+          prm.declare_entry ("Thermal conductivities", "4.7",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of thermal conductivities for background mantle and compositional fields,"
+                             "for a total of N+1 values, where N is the number of compositional fields."
+                             "If only one value is given, then all use the same value. Units: $W/m/K$ ");
+          prm.declare_entry ("Viscosity averaging scheme", "harmonic",
+                             Patterns::Selection("arithmetic|harmonic|geometric|maximum composition"),
+                             "When more than one compositional field is present at a point "
+                             "with different viscosities, we need to come up with an average "
+                             "viscosity at that point.  Select a weighted harmonic, arithmetic, "
+                             "geometric, or maximum composition.");
+          prm.declare_entry ("Elastic shear moduli", "75.0e9",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of elastic shear moduli, $G$, "
+                             "for background material and compositional fields, "
+                             "for a total of N+1 values, where N is the number of compositional fields. "
+                             "The default value of 75 GPa is representative of mantle rocks. Units: Pa.");
+          prm.declare_entry ("Use fixed elastic time step", "false",
+                             Patterns::Bool (),
+                             "Select whether the material time scale in the viscoelastic constitutive"
+                             "relationship uses the regular numerical time step or a separate fixed"
+                             "elastic time step throughout the model run. The fixed elastic time step"
+                             "is always used during the initial time step. Note that there is no"
+                             "physical motivation for using an elastic time step that differs from"
+                             "the numerical time step. As such, this option should only be used in"
+                             "reproducing previous studies or if the numerical time step is a"
+                             "constant value, which can be enforced through the CFL condition and"
+                             "maximum time step parameter. When using a fixed elastic time step that"
+                             "differs the numerical time step it is strongly recommended that the stress"
+                             "averaging scheme is also applied");
+          prm.declare_entry ("Fixed elastic time step", "1.e3",
```

Comment:
Lets think about what exactly this does and then document this parameter a bit more.